src/xdocs/google_style.xml

```diff
@@ -562,7 +562,7 @@
                 </td>
                 <td>
                   <a href="styleguides/google-java-style-20220203/javaguide.html#s3.4.2.1-overloads-never-split">
-                    3.4.2.1 Overloads: never split</a>
+                    3.4.2.1 Overloads Never Split</a>
                 </td>
```

Comment:
had to modify the section name to make XdocsPagesTest happy

Reply:
@romani Are these section names suppose to be a duplicate of the style guide's own names?

If so, I think this means we should change xdoc's logic. Possibly remove non-alphabetic characters, remove spaces, and equals ignoring case.

Reply:
I have fixed this in my other PR: https://github.com/checkstyle/checkstyle/pull/15030#discussion_r1647950400

When it will get merged, I will fix all these edited names. Till then we have to keep it to make tests happy

Reply:
Reverted the section name to the original